### PR TITLE
add dialyzer/dialyxir

### PR DESCRIPTION
**Pull request by niklaslong**
This adds dialyzer to the project. 
We should probably add typespecs + better documentation in a follow-up PR.

### CODE REVIEW

**Comment by niklaslong** [comment]
👍

**Comment by niklaslong** [comment]
@marcoow Finally figured out why `mix dialyzer` was throwing warnings only in CI. This should be all set now.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 language: elixir
-elixir: "1.6.4"
+elixir: "1.6.6"
+otp_release: "21.0"
+
+cache:
+  directories:
+  - _build
```
Comment (marcoow) [inline .travis.yml]: Is this not changing with every Travis build anyway?

File: mix.exs [hunk 2]
```diff
@@ -5,7 +5,10 @@ defmodule Breethe.Umbrella.Mixfile do
     [
       apps_path: "apps",
       start_permanent: Mix.env() == :prod,
-      deps: deps()
+      deps: deps(),
+      dialyzer: [
+        plt_add_apps: [:mix]
```
[the remark this answers is not in the document]
Comment (niklaslong) [reply mix.exs]: It adds the `Mix` app to the plt - avoids throwing a warning for the `Mix.Task` we implemented for syncing data every night.

File: .travis.yml (hunk 1 above)
Comment (niklaslong) [reply .travis.yml]: Yes, but we can cache most of it (it means we don't need to compile the deps every-time for example). In most cases, elixir will automatically update the `_build` directory if there are any changes (like local dev). 

From time to time, we may need to clear the cache (if a build is failing when it shouldn't), but mostly this should speed up our builds dramatically.

It also has the added benefit of including the plt for `dialyzer` (stored in `_build` by default).

File: mix.exs (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcoow) [reply mix.exs]: but what's a plt?

**Pull request merged**